slowquant/unitary_coupled_cluster/sa_ups_wavefunction.py

```diff
@@ -588,6 +588,7 @@ def silent_progress(x: Sequence[float]) -> None:  # pylint: disable=unused-argum
                 self.kappa_redundant[i] = 0
                 self._kappa_redundant_old[i] = 0
         self.thetas = res["x"][param_idx : num_theta + param_idx].tolist()
+        self._sa_energy = res["fun"]
```

Comment:
I think it would be good to have. At this stage it was for me to be able to debug quickly and get some values. So feel free to implement this how it is best for your (new) workflow.

Reply:
I have added it to the optimization branch